api/tasks_test.go

```diff
@@ -638,3 +638,59 @@ func TestSpread_Canonicalize(t *testing.T) {
 		})
 	}
 }
+
+func Test_NewDefaultReschedulePolicy(t *testing.T) {
+	testCases := []struct {
+		inputJobType string
+		expected     *ReschedulePolicy
+	}{
+		{
+			inputJobType: "service",
+			expected: &ReschedulePolicy{
+				Attempts:      intToPtr(0),
+				Interval:      timeToPtr(0),
+				Delay:         timeToPtr(30 * time.Second),
+				DelayFunction: stringToPtr("exponential"),
+				MaxDelay:      timeToPtr(1 * time.Hour),
+				Unlimited:     boolToPtr(true),
+			},
+		},
+		{
+			inputJobType: "batch",
+			expected: &ReschedulePolicy{
+				Attempts:      intToPtr(1),
+				Interval:      timeToPtr(24 * time.Hour),
+				Delay:         timeToPtr(5 * time.Second),
+				DelayFunction: stringToPtr("constant"),
+				MaxDelay:      timeToPtr(0),
+				Unlimited:     boolToPtr(false),
+			},
+		},
+		{
+			inputJobType: "system",
+			expected: &ReschedulePolicy{
+				Attempts:      intToPtr(0),
+				Interval:      timeToPtr(0),
+				Delay:         timeToPtr(0),
+				DelayFunction: stringToPtr(""),
+				MaxDelay:      timeToPtr(0),
+				Unlimited:     boolToPtr(false),
+			},
+		},
+		{
+			inputJobType: "unrecognised",
+			expected: &ReschedulePolicy{
+				Attempts:      intToPtr(0),
+				Interval:      timeToPtr(0),
+				Delay:         timeToPtr(0),
+				DelayFunction: stringToPtr(""),
+				MaxDelay:      timeToPtr(0),
+				Unlimited:     boolToPtr(false),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		actual := NewDefaultReschedulePolicy(tc.inputJobType)
+		assert.Equal(t, tc.expected, actual)
```

Comment:
Not a blocker but wrapping this in a `t.Run(tc.inputJobType, ...)` would make debugging failures a bit faster.